### ISSUE TITLE
Objects inside terrain (medpacks)

### ISSUE THREAD

**Issue by u1ps** [etlegacy/etlegacy#2098]
Grenades disappear under terrain, sometimes landmines are placed in a way that only a tip can be seen, medpacks - sometimes barely top of the object can be seen, hard to notice them on ground.

It seems this happens on terrain that has slopes? (angles)

Example: map radar, medpacks falling inside terrain.
![meddd](https://user-images.githubusercontent.com/99591943/183287531-c93e7753-3a87-4a34-b8ed-aba1fe49637f.png)

**Comment by Aciz** [comment]
This is because the func_group entity in the map thats contains the terrain brushes uses `offset` key for certain shaders, which makes the visual terrain draw N units above the actual collision mesh of the terrain. I don't think there's anything that can be done to this, nor we should, we would just be breaking a feature of q3map2 if we fixed the collision.

**Comment by u1ps** [comment]
map related fix as possibility? This does not happen in all maps. Probably most noticeable in Radar map.

**Comment by Aciz** [comment]
Can be fixed in the map yes, but it requires recompilation I think. But just so you understand, this is not a bug, its an intended thing in Radar.

**Comment by u1ps** [comment]
yes, but i tried it also in non legacy and there the packs did not fall through or follow angle, so i assumed it is some kind of improvement in legacy
![radadaadsadsasd](https://user-images.githubusercontent.com/99591943/183293235-973960f9-f213-4668-8797-5674617822f1.png)

**Comment by Aciz** [comment]
Sorry I totally forgot to follow up on this - I'll try to see what's up with ETPro here this weekend or early next week.

**Comment by Aciz** [comment]
Right, so, ETPro/VET keeps items "level", no matter the orientation of the terrain, where as legacy rotates them so that they lay flush against the terrain plane.

![2022-08-29-201426-radar](https://user-images.githubusercontent.com/14221121/187257891-9a30a7a7-33c0-4796-97fc-491765a39da0.png)

![2022-08-29-201520-radar](https://user-images.githubusercontent.com/14221121/187257979-9138ec3f-f685-4189-8658-ad3fc1397d51.jpg)

This is obviously an aesthetic change on legacy mods side, personally I do like it more than the VET way, but as we can see in Radar's case, it can have a small impact on gameplay. I'm not opposed to reverting this, but I also don't think it is such a big deal. I'll let others make the judgement call here, I'm fine with either option.

**Comment by Aciz** [comment]
Also as I explained, the terrain being non-solid and having different collision box is a feature of the q3map2 compiler, that you can enable for your terrain metashaders while making the map. This is not some bug in legacy mod, it's always been this way in Radar, here's a picture from ETPro near the East Radar.

![2022-08-29-202153-radar](https://user-images.githubusercontent.com/14221121/187260270-52de69e5-f2c9-4731-8c2b-553965d5d6ee.jpg)

**Comment by u1ps** [comment]
yes they seem to go through terrain not just in legacy, But i also noticed if i do it in similar places it seems like it does fall in a little more in legacy. Might make some more pics to confirm. Also players can hide inside terrain when prone, not seeing grenade or landmine etc. If it is more like a map "feature" (bug, whatever is correct to name it) maybe it can be fixed within map or is it too much work whatever is causing it. For some reason it is very noticeable in Radar map.

**Comment by u1ps** [comment]
![teast1212](https://user-images.githubusercontent.com/99591943/187300588-696a6ae5-2278-45f4-b890-f989b9ceaee2.png)
![teast12122](https://user-images.githubusercontent.com/99591943/187300668-14c5c0bf-9b2c-4e6d-8ad1-39a4eefefb69.png)

**Comment by Aciz** [comment]
This is again due to legacy aligning the medpack with the terrain plane. It doesn't sink any more than ETPro/VET, it just seems that way since it's slightly tilted.

![image](https://user-images.githubusercontent.com/14221121/187306314-2f4c7582-e865-4fed-8936-58dc136cc601.png)
![image](https://user-images.githubusercontent.com/14221121/187306335-49f2e05c-238e-4114-bf67-77a4f077bc7f.png)
![image](https://user-images.githubusercontent.com/14221121/187306375-be8eb43a-cf0f-4ace-9037-cc743f2e8547.png)

![image](https://user-images.githubusercontent.com/14221121/187306555-b2e20cee-a634-4c65-ab5a-57ecab03951d.png)
![image](https://user-images.githubusercontent.com/14221121/187306576-fb2a2ea7-9147-40bf-962c-a5ff73b93619.png)
![image](https://user-images.githubusercontent.com/14221121/187306602-1d7827f8-e4a0-46c6-85f4-3f2979454bf5.png)

**Comment by Aciz** [comment]
> If it is more like a map "feature" (bug, whatever is correct to name it) maybe it can be fixed within map or is it too much work whatever is causing it. For some reason it is very noticeable in Radar map.

This would require for us to ship our own version of Radar, the terrain shift happens during the map compilation, we cannot patch the one in pak0.pk3 with a shader tweak.

**Comment by u1ps** [comment]
Wonder if it is possible to make objects appear higher with offset when interacting with this shade?

**Comment by Aciz** [comment]
That would be the best solution for this definitely, but I'm not sure if it's possible, need to investigate.

**Comment by x0rnn** [comment]
If etl_radar is made, also fix that microscopic ledge thing at main gate that you get stuck at if you don't walk fast enough please :D

**Comment by ryzyk-krzysiek** [comment]
> If etl_radar is made, also fix that microscopic ledge thing at main gate that you get stuck at if you don't walk fast enough please :D

It kind of is fixed though https://github.com/etlegacy/etlegacy/issues/1590 https://github.com/etlegacy/etlegacy/issues/516

**Comment by x0rnn** [comment]
Oh cool, thought map needed to be recompiled for that!

**Comment by u1ps** [comment]
> That would be the best solution for this definitely, but I'm not sure if it's possible, need to investigate.

I would close this as map issue if that investigation results as not possible :D
